### PR TITLE
Add .Net Auto-instrumentation Developer Documentation

### PR DESCRIPTION
**Pull request by jvoravong**
**Description:** 
- Add support developer documentation for Splunk .Net auto-instrumentation in Kubernetes.
- The mentioned .NET demo will be added in a follow up PR.
- Adding tests in a follow up PR to keep the PR diff low.

**Testing:** <Describe what testing was performed and which tests were added.>
- Adding functional tests in a follow up PR.

**Documentation:** <Describe the documentation added.>
- Added documentation for how to use .Net auto-instrumentation. [Customer documentation](https://docs.splunk.com/observability/en/gdi/opentelemetry/auto-instrumentation/auto-instrumentation-dotnet-k8s.html) has been created based off of initial changes and feedback.

### CODE REVIEW

**Comment by atoulme** [comment]
looks like a great start!

File: examples/enable-operator-and-auto-instrumentation/otel-demo-dotnet.md [hunk 1]
```diff
@@ -0,0 +1,252 @@
+# Example of chart configuration
+
+## How to deploy the OpenTelemetry Operator and .Net auto-instrumentation
```
Comment (pjanotti) [inline examples/enable-operator-and-auto-instrumentation/otel-demo-dotnet.md:3]: nit: the brand casing is `.NET`

Comment (jvoravong) [reply examples/enable-operator-and-auto-instrumentation/otel-demo-dotnet.md]: Done

File: functional_tests/testdata/dotnet/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# .Net test image
+
+This image is used for testing the auto-instrumentation of .Net application through the OpenTelemetry Operator.
+
+This image is pushed to https://quay.io/repository/splunko11ytest/dotnet_test.
+
+The container performs two separate functions:
+* It runs a .Net HTTP server on port 3000 of the container host.
```
Comment (Kielek) [inline functional_tests/testdata/dotnet/README.md:8]:
```suggestion
# .NET test image

This image is used for testing the auto-instrumentation of .NET application through the OpenTelemetry Operator.

This image is pushed to https://quay.io/repository/splunko11ytest/dotnet_test.

The container performs two separate functions:
* It runs a .NET HTTP server on port 3000 of the container host.
```

File: examples/enable-operator-and-auto-instrumentation/otel-demo-dotnet.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+**Note:**
+- This will cause the dotnet-test pod to restart.
+- The annotation value "default/splunk-otel-collector" refers to the Instrumentation configuration named `splunk-otel-collector` in the `default` namespace.
+- If the chart is not installed in the "default" namespace, modify the annotation value to be "{chart_namespace}/splunk-otel-collector".
+- For troubleshooting purposes, examining the logs located in /var/log/opentelemetry/dotnet within the instrumented pod can provide valuable debugging insights.
+
+Remove the annotations to disable instrumentation:
+
+```bash
+kubectl patch deployment dotnet-test -n dotnet-demo --type=json -p='[{"op": "remove", "path": "/spec/template/metadata/annotations/instrumentation.opentelemetry.io~otel-dotnet-auto-runtime"}]'
```
Comment (asuresh4) [inline examples/enable-operator-and-auto-instrumentation/otel-demo-dotnet.md:111]:
```suggestion
kubectl patch deployment dotnet-test -n dotnet-demo --type=json -p='[{"op": "remove", "path": "/spec/template/metadata/annotations/instrumentation.opentelemetry.io~1otel-dotnet-auto-runtime"}]'
```

File: examples/enable-operator-and-auto-instrumentation/otel-demo-dotnet.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+The .NET demo will create a dotnet-demo namespace and deploys the related .NET application to it.
+If you have your own .NET application you want to instrument, you can still use the steps below as an example for how
+to instrument your application.
+
+```bash
+kubectl create namespace dotnet-demo
+```
+
+```bash
+curl https://raw.githubusercontent.com/signalfx/splunk-otel-collector-chart/main/examples/enable-operator-and-auto-instrumentation/dotnet-demo/deployment.yaml | kubectl apply -n otel-demo -f -
```
Comment (asuresh4) [inline examples/enable-operator-and-auto-instrumentation/otel-demo-dotnet.md:19]:
```suggestion
curl https://raw.githubusercontent.com/signalfx/splunk-otel-collector-chart/main/examples/enable-operator-and-auto-instrumentation/dotnet-demo/deployment.yaml | kubectl apply -n dotnet-demo -f -
```

File: functional_tests/testdata/dotnet/README.md (hunk 2 above)
Comment (jvoravong) [reply functional_tests/testdata/dotnet/README.md:8]: Done

File: examples/enable-operator-and-auto-instrumentation/otel-demo-dotnet.md (hunk 4 above)
Comment (jvoravong) [reply examples/enable-operator-and-auto-instrumentation/otel-demo-dotnet.md:19]: Done

**Comment by jvoravong** [comment]
This PR was originally going to add documentation, a demo application, and functional testing. However, do to scope of work increasing, I'm splitting up this work. I'm only adding documentation in this PR. The demo application and functional tests will be added in separate followup PRs.

File: examples/enable-operator-and-auto-instrumentation/otel-demo-dotnet.md (hunk 3 above)
Comment (jvoravong) [reply examples/enable-operator-and-auto-instrumentation/otel-demo-dotnet.md:111]: > Done

**Review by Kielek — APPROVED** [review]

**Review by samiura — APPROVED** [review]

**Pull request merged**